sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -578,27 +573,35 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         AppendData.byName(v2Relation, df.logicalPlan, extraOptions.toMap)
 
       case (SaveMode.Overwrite, _) =>
-        ReplaceTableAsSelect(
-          catalog,
-          ident,
-          partitioningAsV2,
+        ReplaceTableAsSelectStatement(
+          nameParts,
           df.queryExecution.analyzed,
-          Map(TableCatalog.PROP_PROVIDER -> source) ++ getLocationIfExists,
+          partitioningAsV2,
+          None,
```

Comment:
Not related to this PR, but do we plan to support buckets for datasource V2?

Reply:
Yes. I have a design doc I can share with you.

Reply:
Great, thanks!